pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -237,12 +250,35 @@ var _ framework.PreScorePlugin = &dynamicResources{}
 var _ framework.ReservePlugin = &dynamicResources{}
 var _ framework.EnqueueExtensions = &dynamicResources{}
 var _ framework.PostBindPlugin = &dynamicResources{}
+var _ framework.RequeuePlugin = &dynamicResources{}
 
 // Name returns name of the plugin. It is used in logs, etc.
 func (pl *dynamicResources) Name() string {
 	return Name
 }
 
+func (pl *dynamicResources) Requeue(ctx context.Context, p *v1.Pod, event *framework.ClusterEvent, oldObj, obj interface{}) framework.SchedulingHint {
+	if !event.In(pl.EventsToRegister()) {
+		// not interested event.
+		return framework.PodNotAffected
+	}
+
+	switch event.Resource {
+	case framework.ResourceClaim:
+		return pl.isSchedulableAfterClaimChange(oldObj, obj, p)
```

Comment:
There is a mismatch between `oldObj` and `obj` and what `isSchedulableAfterClaimChange` expects: my hint function API has separate `oldObj` and `newObj` parameters, so there's no ambiguity where to find a deleted object (in `oldObj`).

`Requeue` is different and passes the deleted object in `obj`.

I prefer `oldObj` and `newObj` because I find it easier to read and understand what is what - just my two cents, both approaches work.

However, the calls to `isSchedulableAfterClaimChange` and `isSchedulableAfterPodSchedulingChange` need to be fixed to match the `Requeue` API.

Reply:
FWIW, `MoveAllToActiveOrBackoffQueue` also uses the `oldObj` + `newObj` approach.